cmd/docker/docker.go

```diff
@@ -194,9 +197,46 @@ func tryPluginRun(dockerCli command.Cli, cmd *cobra.Command, subcommand string,
 	}
 	plugincmd.Env = append(envs, plugincmd.Env...)
 
+	// TODO(laurazard): this will break if we run this plugin more than once,
+	// as we will collide with the already-in-use socket name. We should retry
+	// a couple of times and add a counter or some other unique element to them
+	// socket name
+	pluginCloseSocketAddr := net.UnixAddr{Name: "@docker_cli_" + subcommand, Net: "unix"}
```

Comment:
@thaJeztah @vvoland @neersighted thoughts?

Reply:
Are there any constraints on these? i.e., could we just a GUID (`docker_cli_<GUID>`)?

Reply:
It's an arbitrary string that has to be globally unique; indeed a UUID or similar (e.g. a mktemp template) should be just fine.

Reply:
I found https://github.com/distribution/distribution/blob/main/uuid/uuid.go (which we're already importing) which results in sockets that look like
```console
☆｡･:*:･ﾟ⭑ laurabrehm ✿ cli [plugin-signal-handling] ⭑ ss -ax | grep docker_cli
u_str ESTAB  0      0                                        @docker_cli_c264cbff-49e2-4cab-912f-175481090216 276544            * 274216
u_str LISTEN 0      4096                                     @docker_cli_c264cbff-49e2-4cab-912f-175481090216 273236            * 0
```

which looks fine for me, but I do wonder if that goes over some arbitrary character limit in any platform.

Reply:
Per `unix(7)`, the `sun_path` field is 108 bytes long. Post https://github.com/golang/go/issues/21965, we should have all 108 bytes to work with.

Other platforms are likely compatible with this length limit, as it's been in place since the early versions of Berkley sockets as best I can tell. We could add some tests for max length to make sure this works across all platforms, if desired.